ui/api-client/src/runtime.ts

```diff
@@ -422,7 +422,8 @@ export class JSONApiResponse<T> {
     constructor(public raw: Response, private transformer: ResponseTransformer<T> = (jsonValue: any) => jsonValue) {}
 
     async value(): Promise<T> {
-        return this.transformer(await this.raw.json());
+        const response = await this.raw.json();
+        return this.transformer(response.data);
     }
```

Comment:
This is the same thing the middleware was doing but for all responses. Letting the client handle it means this will only be used for responses that are `JSONApiResponse` type and not for `VoidResponse`.